symphony-bdk-core/src/main/java/com/symphony/bdk/core/auth/impl/BotAuthenticatorCertImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.symphony.bdk.core.auth.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.auth.BotAuthenticator;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.gen.api.CertificateAuthenticationApi;
+import com.symphony.bdk.gen.api.model.Token;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Bot authenticator certificate implementation.
+ *
+ * @see <a href="https://developers.symphony.com/symphony-developer/docs/bot-authentication-workflow-1">BotAuthentication Workflow</a>
+ */
+@Slf4j
+@API(status = API.Status.INTERNAL)
+public class BotAuthenticatorCertImpl implements BotAuthenticator {
+
+  private final ApiClient sessionAuthClient;
+  private final ApiClient keyAuthClient;
+
+  public BotAuthenticatorCertImpl(@Nonnull ApiClient sessionAuthClient,
+      @Nonnull ApiClient keyAuthClient) {
+    this.sessionAuthClient = sessionAuthClient;
+    this.keyAuthClient = keyAuthClient;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public AuthSession authenticateBot() throws AuthUnauthorizedException {
+    AuthSessionCertImpl authSession = new AuthSessionCertImpl(this);
+    authSession.refresh();
+    return authSession;
+  }
+
+  protected @Nonnull String retrieveSessionToken() throws AuthUnauthorizedException {
+    log.debug("Start retrieving sessionToken using certificate authentication...");
+    return doRetrieveToken(this.sessionAuthClient);
+  }
+
+  protected  @Nonnull String retrieveKeyManagerToken() throws AuthUnauthorizedException {
+    log.debug("Start retrieving keyManagerToken using certificate authentication...");
+    return doRetrieveToken(this.keyAuthClient);
+  }
+
+  private String doRetrieveToken(ApiClient client) throws AuthUnauthorizedException {
+    try {
+      final Token token = new CertificateAuthenticationApi(client).v1AuthenticatePost();
+      log.debug("{} successfully retrieved.", token.getName());
+      return token.getToken();
+    } catch (ApiException ex) {
+      if (ex.getCode() == 401) {
```

Comment:
for all http error we must use this class provided by java : https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html#HTTP_UNAUTHORIZED